python/src/iceberg/io/base.py

```diff
@@ -41,10 +41,10 @@ class InputStream(Protocol):  # pragma: no cover
     implementation's `open(...)` method. These methods are a subset of IOBase/RawIOBase.
     """
 
-    def read(self, size: int) -> bytes:
+    def read(self, n: int = -1) -> bytes:
```

Comment:
I used that to be consistent with the python IO built-ins. The value is described here: https://docs.python.org/3/library/io.html#io.RawIOBase.read

> Read up to size bytes from the object and return them. As a convenience, if size is unspecified or -1, all bytes until EOF are returned. Otherwise, only one system call is ever made. Fewer than size bytes may be returned if the operating system call returns fewer than size bytes.

Reply:
Sound good. We may want to consider adding docs that at least point to the canonical ones, but that's not something that should block this PR.